### PR TITLE
Fix trigger timestamp (SpidrTrigger and TLU fine)

### PR DESCRIPTION
**Pull request by jekroege**
CLICTD Event Converter:
 - place timestamp between shutter open and shutter close

CLICpix2 Event Converter:
 - same

Timepix3 Event Converter:
 - add detector type "SpidrTrigger"
 - add dummy plane containing dummy pixel with trigger input timestamp
 - add explanation of data format

TLU Event Converter:
 - add tags with fine timestamps

euCliReader:
 - correct cout of fine timestamps

### CODE REVIEW

File: user/caribou/module/src/CLICTDEventConverter.cc [hunk 1]
```diff
@@ -223,8 +223,8 @@ bool CLICTDEvent2StdEventConverter::Converting(eudaq::EventSPC d1, eudaq::Standa
       tot = 1;
     }
 
-    // Time defaults ot rising shutter edge:
-    auto timestamp = shutter_open;
+    // Time defaults to center between rising and falling shutter edge:
```
Comment (simonspa) [inline user/caribou/module/src/CLICTDEventConverter.cc]: I somehow remember fixing this already. Hm...

File: user/timepix3/module/src/Timepix3Event2StdEventConverter.cc [hunk 2]
```diff
@@ -56,11 +66,23 @@ bool Timepix3TrigEvent2StdEventConverter::Converting(eudaq::EventSPC ev, eudaq::
...
+  uint64_t triggerTime = timestamp * 3125 +(stamp * 3125) / 12;
 
   // Set timestamps for StdEvent in nanoseconds (timestamps are picoseconds):
   d2->SetTimeBegin(triggerTime);
   d2->SetTimeEnd(triggerTime);
+
+  // Identify the detetor type
+  d2->SetDetectorType("SpidrTrigger");
+
+  // Create a StandardPlane representing one DUMMY sensor plane
```
Comment (simonspa) [inline user/timepix3/module/src/Timepix3Event2StdEventConverter.cc]: Do we need this really? Feels odd.

**Comment by simonspa** [comment]
Ah, could you file this against `caribou`?

**Review by simonspa — CHANGES_REQUESTED** [review]
Looks good in general, see two comments below.

File: user/timepix3/module/src/Timepix3Event2StdEventConverter.cc (hunk 2 above)
Comment (jekroege) [reply user/timepix3/module/src/Timepix3Event2StdEventConverter.cc]: I had no other idea how to ship this info into corry.

File: user/caribou/module/src/CLICTDEventConverter.cc (hunk 1 above)
Comment (jekroege) [reply user/caribou/module/src/CLICTDEventConverter.cc]: I just realized, this whole thing is the same as MR !11

File: user/tlu/exe/src/euCliTluReader.cxx [hunk 3]
```diff
@@ -118,27 +118,27 @@ int main(int /*argc*/, const char **argv) {
...
-          auto finets2 = ev->GetTag("FINE_TS2", "NAN");
-          auto finets3 = ev->GetTag("FINE_TS3", "NAN");
-          auto finets4 = ev->GetTag("FINE_TS4", "NAN");
-          auto finets5 = ev->GetTag("FINE_TS5", "NAN");
-          std::cout << runNumber << "," << 
-      		   eventNumber << "," << 
-		   triggerNumber << "," << 
-		   timeStampBegin << "," << 
-		   timeStampEnd << "," << 
+          auto scaler0 = subev->GetTag("SCALER0", "NAN");
```
Comment (simonspa) [inline user/tlu/exe/src/euCliTluReader.cxx:121]: can't we just

```
d2->SetTag("FINE_TS0", d1->GetTag("FINE_TS0", "NAN"));
```

Or just keep it, might be cleaner. Is there a reason these fine TS are not used to calculate the actual time more finely than the 40MHz?

File: user/tlu/exe/src/euCliTluReader.cxx [hunk 4]
```diff
@@ -118,27 +118,27 @@ int main(int /*argc*/, const char **argv) {
 	  auto timeStampEnd = subev->GetTimestampEnd();
           auto particles = subev->GetTag("PARTICLES", "NAN");
 	  auto triggersFired = subev->GetTag("TRIGGER" , "NAN");
```
Comment (simonspa) [inline user/tlu/exe/src/euCliTluReader.cxx:120]: unused variable?

**Review by simonspa — CHANGES_REQUESTED** [review]

File: user/timepix3/module/src/Timepix3Event2StdEventConverter.cc (hunk 2 above)
Comment (jekroege) [reply user/timepix3/module/src/Timepix3Event2StdEventConverter.cc]: Don't remember why I did it in the first place, the timestamps are available as the event timestamps anyway:
```
d2->SetTimeBegin(triggerTime);
```
Reverted this change.

File: user/tlu/exe/src/euCliTluReader.cxx (hunk 4 above)
Comment (jekroege) [reply user/tlu/exe/src/euCliTluReader.cxx:120]: It's used in line 139.

File: user/tlu/exe/src/euCliTluReader.cxx (hunk 3 above)
Comment (jekroege) [reply user/tlu/exe/src/euCliTluReader.cxx:121]: The reason is that the 40 MHz timestamp is the "coincidence time" the TLU determines internally.

The four fine timestamps are those of the individual trigger inputs, so it depends on which of them are actually used, e.g. if we have trg0 and trg1 connected, the (trg0+trg1)/2 would be the best estimate but trg2 and trg3 will always be zero.
EUDAQ2 is agnostic to how the trigger inputs are connected.

Comment (jekroege) [reply user/tlu/exe/src/euCliTluReader.cxx:121]: 2nd point: 
```
d2->SetTag("FINE_TS0", d1->GetTag("FINE_TS0", "NAN"));
```
would be a bit shorter, but I find it more aestetic as is. Do you have a strong opinion on that? :-D

Comment (simonspa) [reply user/tlu/exe/src/euCliTluReader.cxx:121]: Nah, it's alright.

When you say "EUDAQ2 is agnostics" who isn't then? How do you calculate this downstream?

Comment (jekroege) [reply user/tlu/exe/src/euCliTluReader.cxx:121]: Currently, we're only using the coarse timestamp (as before).

We didn't decide on a more clever way if implementing/configuring this yet.

Comment (simonspa) [reply user/tlu/exe/src/euCliTluReader.cxx:121]: Alright. As soon as we do, this would be the place to do so I guess...

Comment (jekroege) [reply user/tlu/exe/src/euCliTluReader.cxx:121]: I agree, then we'd either need to implement some logic like "average over all which are non-zero" (but they are zero correctly sometimes) or make it configurable.

Let's postpone this discussion until we actually need it.

**Pull request merged**